uefi/src/proto/debug/mod.rs

```diff
@@ -170,3 +170,46 @@ pub enum ProcessorArch: u32 => {
     /// RISC-V 128-bit
     RISCV_128   = 0x5128,
 }}
+
+/// The debug port protocol abstracts the underlying debug port
+/// hardware, whether it is a regular Serial port or something else.
+#[repr(C)]
+#[unsafe_protocol("eba4e8d2-3858-41ec-a281-2647ba9660d0")]
+pub struct DebugPort {
+    reset: extern "efiapi" fn(this: &DebugPort) -> Status,
+    write: extern "efiapi" fn(
+        this: &DebugPort,
+        timeout: u32,
+        buffer_size: &mut usize,
+        buffer: *const c_void,
+    ) -> Status,
+    read: extern "efiapi" fn(
+        this: &DebugPort,
+        timeout: u32,
+        buffer_size: &mut usize,
+        buffer: *mut c_void,
+    ) -> Status,
+    poll: extern "efiapi" fn(this: &DebugPort) -> Status,
+}
+
+impl DebugPort {
+    /// Flushes the port.
+    pub fn reset(&self) -> Result {
+        (self.reset)(self).into()
+    }
+
+    /// Writes 8 bytes at a time to the underlying SerialIo protocol.
+    pub fn write(&self, timeout: u32, buffer_size: &mut usize, buffer: *const c_void) -> Result {
```

Comment:
For the public read/write interface, let's use a slice parameter. See here for an example: https://github.com/rust-osdev/uefi-rs/blob/main/uefi/src/proto/console/serial.rs#L96